### PR TITLE
Refactor device TPM logic to better support Linux

### PR DESCRIPTION
**Pull request by codingllama**
Split the current Windows TPM implementation into generic and OS-specific parts, providing a simple framework to add Linux on top of it.

First part of Linux TPM support.

https://github.com/gravitational/teleport.e/issues/827

### CODE REVIEW

**Comment by codingllama** [comment]
This is much smaller than it looks - I suggest reviewing commit by commit.

* The first commit does `cp device_windows.go tpm_windows.go` (9bcc7d1)
* The second commit re-arranges (via deleting) device_windows.go and tpm_windows.go so the former has only Windows-specific parts, while the latter has generic parts (0b3e30b)
* The third commit actually does some work, adding the tpmDevice struct and dropping the implied "_windows" build tag from tpm_windows.go (by renaming the file). This is where we get actual code changes. (3199a97)
* The forth commit makes my life easier by ignoring my bogus "reported_asset_tag" (c046923)

**Comment by codingllama** [comment]
FYI @strideynet, in case you have some spare review cycles.

File: lib/devicetrust/native/tpm_device.go [hunk 1]
```diff
@@ -1,3 +1,5 @@
+//go:build windows
```
Comment (codingllama) [inline lib/devicetrust/native/tpm_device.go:1]: I'll add linux to it when the time comes.

**Review by zmb3 — APPROVED** [review]

**Review by strideynet — APPROVED** [review]
Looks good - I think this is what we had in mind originally.

**Comment by codingllama** [comment]
Thanks for the quick reviews!

**Pull request merged**